### ISSUE TITLE
Move weaveworks/common back to upstream

### ISSUE THREAD

**Issue by cyriltovena** [grafana/loki#945]
To not block some PR we moved weaveworks/common to a personal fork. Once weaveworks/common#153 is merged we can go back to upstream.

**Comment by ThoreKr** [comment]
As weaveworks/common#153 is more or less dead and waiting for a rebase I reopened a new (rebased) PR (weaveworks/common#167).

**Comment by clickyotomy** [comment]
Hello! Can I work on this after https://github.com/weaveworks/common/pull/167 is merged?

**Comment by ThoreKr** [comment]
@clickyotomy If you are more familiar with the dependencies a bit of help in weaveworks/common#167 would be most welcome.

Initially it was only meant to be a rebase to resolve a merge conflict, but now the actual change has been reviewed again.

**Comment by ThoreKr** [comment]
@clickyotomy Please also take a look at my notes in #1051 as there is apparently another version in the `vendor` subdirectory.

**Comment by clickyotomy** [comment]
@ThoreKr, I am not familiar with the changes introduced in https://github.com/weaveworks/common/pull/167, but I will try to help. :)
  * I don't know what needs to be done [here](https://github.com/weaveworks/common/pull/167/files#r329560591) - should we include/replace `localhost` on all the tests?
  * However, [here](https://github.com/weaveworks/common/pull/167/files#r329560936) - maybe we can get rid of `HTTPListen{Port,Address}` for `TestErrorInstrumentationMiddleware`, because it's probably not needed for that test.

How do I make changes to the PR? Fork from https://github.com/ThoreKr/weaveworks-common/tree/server-listen-addr and open a new one?

Regarding https://github.com/grafana/loki/issues/1051, this repo was moved to `go modules` (https://github.com/grafana/loki/pull/1062) for dependency management. We might just have to do a `go mod edit -dropreplace=personal_fork[@v]` and use the latest from upstream for this issue.

**Comment by ThoreKr** [comment]
@clickyotomy Me neither, I just rebased it. Maybe wait for comments from upstream whether it should be tested or not.

**Comment by ThoreKr** [comment]
Still waiting for weaveworks-common.

Starting to get annoyed.

**Comment by ThoreKr** [comment]
@clickyotomy it's merged!
